CONTRIBUTING.md

```diff
@@ -155,6 +160,32 @@ Please follow these formatting guidelines:
  * IntelliJ: `Preferences/Settings->Editor->Code Style->Java->Imports`. There are two configuration options: `Class count to use import with '*'` and `Names count to use static import with '*'`. Set their values to 99999 or some other absurdly high value.
 * Don't worry too much about import order. Try not to change it but don't worry about fighting your IDE to stop it from doing so.
 
+### License Headers
+
+We require license headers on all Java files. You will notice that all the Java files in
+the top-level `x-pack` directory contain a separate license from the rest of the repository. This
+directory contains commercial code that is associated with a separate license. It can be helpful
+to have the IDE automatically insert the appropriate license header depending which part of the project
+contributions are made to.
+
+#### Intellij: Copyright & Scope Profiles
+
+To have Intellij insert the correct license, it is necessary to create to copyright profiles.
+These may potentially be called `apache2` and `commercial`. These can be created in
```

Comment:
(nitpick) I think it's a bit clearer to use the names `core` and `xpack` for the copyright + scope profiles (especially for the scopes, which may be useful in other contexts).

Reply:
I think the existing names `apache2` and `commercial` are correct. They are the license names (I personally prefer `commercial` over `elastic`, but the latter would also be appropriate, since it is more technically accurate). But these are just suggestions anyways. :)

Reply:
I was hoping to keep the example names as close to the license/copyright they are representing. I think I will leave it to the contributor to decide which name to choose